src/sfdp.jl

```diff
@@ -106,8 +110,8 @@ function Base.iterate(iter::LayoutIterator{<:SFDP}, state)
         if any(isnan, force)
             # if two points are at the exact same location
             # use random force in any direction
-            rng = MersenneTwister(algo.seed + i)
-            force = randn(rng, Ftype)
+            rng = copy(algo.rng)
```

Comment:
Can you add a comment why we need a copy here?

Reply:
I did now. The reason beeing is that `algorithm(g)` should return the same layout again and again. Therefore we should never advance the rng "prototype" from the algorithm struct but instead create a copy if we need something random.